src/db.rs

```diff
@@ -2289,47 +2290,17 @@ impl<T: ThreadMode, D: DBInner> DBCommon<T, D> {
     /// and end key
     pub fn live_files(&self) -> Result<Vec<LiveFile>, Error> {
         unsafe {
-            let files = ffi::rocksdb_livefiles(self.inner.inner());
-            if files.is_null() {
+            let livefiles_ptr = ffi::rocksdb_livefiles(self.inner.inner());
+            if livefiles_ptr.is_null() {
                 Err(Error::new("Could not get live files".to_owned()))
             } else {
-                let n = ffi::rocksdb_livefiles_count(files);
-
-                let mut livefiles = Vec::with_capacity(n as usize);
-                let mut key_size: usize = 0;
-
-                for i in 0..n {
-                    let column_family_name =
-                        from_cstr(ffi::rocksdb_livefiles_column_family_name(files, i));
-                    let name = from_cstr(ffi::rocksdb_livefiles_name(files, i));
-                    let size = ffi::rocksdb_livefiles_size(files, i);
-                    let level = ffi::rocksdb_livefiles_level(files, i);
-
-                    // get smallest key inside file
-                    let smallest_key = ffi::rocksdb_livefiles_smallestkey(files, i, &mut key_size);
-                    let smallest_key = raw_data(smallest_key, key_size);
-
-                    // get largest key inside file
-                    let largest_key = ffi::rocksdb_livefiles_largestkey(files, i, &mut key_size);
-                    let largest_key = raw_data(largest_key, key_size);
-
-                    livefiles.push(LiveFile {
-                        column_family_name,
-                        name,
-                        size,
-                        level,
-                        start_key: smallest_key,
-                        end_key: largest_key,
-                        num_entries: ffi::rocksdb_livefiles_entries(files, i),
-                        num_deletions: ffi::rocksdb_livefiles_deletions(files, i),
-                    });
-                }
```

Comment:
I've factored out this block to a static fn in the LiveFile impl so that I can reuse it for `ExportImportFilesMetaData`.